### ISSUE TITLE
Enable/Disable Filters Fails if Model hasn't been created.

### ISSUE THREAD

**Issue by thisispaulsmith** [jcachat/EntityFramework.DynamicFilters#34]
OnModelCreating (where filters are defined) isn't called until a context is first used so if you enable/disable or set a scoped variable on a filter before query execution an error is thrown.

Am I missing something?

**Comment by thisispaulsmith** [comment]
Apologies I've just seen similar closed issue (#24)

It would be nice if the filter key could be added (if it doesn't exist) during enabling/disabling as it is in EntityFramework.Filters. This would remove the need for calling `Database.Intialize(false)`

**Comment by jcachat** [comment]
When enabling/disabling filters, I think I can allow that.  But when setting a parameter, I need information about how the filter was created - specifically, I need the parameter names so that I know how to store the parameter values correctly in cases where there are multiple parameters.

I'll take a look at this and try to make some improvements.

**Comment by thisispaulsmith** [comment]
It's not essential. Just makes the code cleaner as it's not immediately clear why you're calling `Database.Initialize(false)` without a dirty comment.

**Comment by jcachat** [comment]
jwillmer's comment on #36 got me thinking.  I dug into the EF code a bit on how context.Database.Initialize works.  Turns out it's a very lightweight call to make if the database is already initialized.  So calling it as he suggested works just fine.

I added a call to that into all of the public extension methods that rely on filters being configured - specially all of the enable, disable, and set param methods.  And I created a full set of unit tests to verify that all works correctly.

Changes are uploaded to git and NuGet.
